src/querystring.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{error::Error, Context};
+use http::StatusCode;
+use serde::Deserialize;
+
+/// An extension trait for `Context`, providing query string deserialization.
+pub trait ExtractQuery<'de> {
+    fn query<T: Deserialize<'de>>(&'de self) -> Result<T, Error>;
```

Comment:
Should this be `query_string` instead? I would find it confusing, in the land of web, and database queries.

Reply:
I opted for `query` because I was concerned about the opposite: that users would expect Rust methods ending in  `_string` methods might return `String`.

I'd like to hear more feedback on this question, though, because I don't mind either way.

Reply:
I don't really think that's the rust convention. But since you mentioned, I just ran this on the crates registry, just to see for myself. 

`find . -type f -name "*.rs" -exec grep "_string(" /dev/null {} \;`

Unless the prefix part of the `string` has something that indicates `to_`/`as_`/`gen_`, or any other explicit notion that it returns a `String`, I don't think there was any. Languages like Go use that convention, but I doubt rust does.

Reply:
Actually, I'm curious now. Let's invoke the big guns - @aturon, thoughts? (also if I'm in the vicinity of being right, that just brings about more questions - `body_as_string`/`body_as_json` etc?

Reply:
Agree this might be confusing -- "query string" might be indeed interpreted as: "returns a string". Without reading the code I wouldn't know what type it returns.

Also "query" might be confusing the moment we start hooking up databases to context. One could reasonably expect that the `query` method allows you to construct a database query.

I'm seeing a few other naming options:
- `url_query`, similar to [`UrlQuery`](https://docs.rs/url/1.7.2/url/struct.UrlQuery.html), making it clear it's the query portion of the URL
- `search` or `search_params` as in url search params, which [is used by node.js](https://nodejs.org/api/url.html#url_url_search) and part of [the WhatWG url spec](https://url.spec.whatwg.org/#url-class), but is less used and doesn't have precedent in the `url` crate.

Reply:
+1 for @yoshuawuyts `url_query`